### PR TITLE
TagBox: add possibility to change caret position by mousedown (T1046705)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

**Review by alexander-kotov-dx — DISMISSED** [review]

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 1]
```diff
@@ -487,12 +487,14 @@ const DropDownList = DropDownEditor.inherit({
         this._setListFocusedElementOptionChange();
     },
 
-    _renderPreventBlur: function($target) {
+    _renderPreventBlur: function($target, exclusionElements = []) {
```
Comment (ksercs) [inline js/ui/drop_down_editor/ui.drop_down_list.js]: Isn't it better to implement 2 different functions: one for ddList and another for  TagBox?

Why I think it's a good idea:
- second parameter is passed from inheritor to parent component (and it's useful only for 1/4 inheritors)
- new specific case is actual only for the TagBox
- we create event for TagBox on ddList namespace now

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 2]
```diff
@@ -5846,6 +5847,19 @@ QUnit.module('single line mode', {
 
         $inputWrapper.trigger('mousedown');
     });
+
+    QUnit.test('mousedown should not be prevented when input field clicked (T1046705)', function(assert) {
+        assert.expect(1);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Why should ew use `expect`? Is there really some async code execution?

Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Nope 
Just copypasted previus test whithout check as always, sry

File: js/ui/drop_down_editor/ui.drop_down_list.js (hunk 1 above)
Comment (Zedwag) [reply js/ui/drop_down_editor/ui.drop_down_list.js]: done

**Review by ksercs — APPROVED** [review]

**Pull request merged**